piet/src/text.rs

```diff
@@ -30,15 +36,76 @@ pub trait Text: Clone {
     fn new_text_layout(&mut self, text: &str) -> Self::TextLayoutBuilder;
 }
 
-/// A representation of a font.
+/// A reference to a font family.
+///
+/// This may be either a CSS-style "generic family name", such as "serif"
+/// or "monospace", or it can be an explicit family name.
 ///
-/// This type may not be an actual font object, but is some type that can
-/// be resolved to a concrete font.
+/// To use a generic family name, use the provided associated constants:
+/// `FontFamily::SERIF`, `FontFamily::SANS_SERIF`, `FontFamily::SYSTEM_UI`,
+/// and `FontFamily::MONOSPACE`.
 ///
-/// When loading a system font, this type can be thought of as a font *family*;
-/// if you change the weight or the style in a layout span, that may cause a
-/// different font in that family to be used for the actual drawing.
-pub trait Font: Clone {}
+/// To use a specific font family you should not construct this type directly;
+/// instead you should verify that the desired family exists, via the
+/// [`Text::font`] API.
+///
+/// [`Text::font`]: trait.Text.html#tymethod.font
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FontFamily(FontFamilyInner);
+
+/// The inner representation of a font family.
+///
+/// This is not public API for users of piet; it is exposed for backends only.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[doc(hidden)]
+#[non_exhaustive]
```

Comment:
I'm not sure non_exhaustive is appropriate here; the back-end should probably handle all cases. But since a default fallback font is reasonable, I'm ok with it.

Reply:
I think my reasoning was that since this is technically public api, we wanted to indicate that more variants might be added. 🤷